### PR TITLE
dibs: implement binary-search-tree

### PR DESCRIPTION
**Pull request by laurauzcategui**
I will implement binary-search-tree in : [Binary Search Tree - X-Common](https://github.com/exercism/x-common/tree/master/exercises/binary-search-tree)

### CODE REVIEW

**Comment by kytrinyx** [comment]
Thanks so much for working on this :heart:

It looks like you're still working on getting the test suite implemented. If you want to have a look at some of the other implementations of this exercise, you can check out:

- https://github.com/exercism/xcsharp/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xclojure/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xcoffeescript/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xecmascript/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xfsharp/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xgo/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xhaskell/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xjavascript/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xlua/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xperl5/tree/master/binary-search-tree
- https://github.com/exercism/xruby/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xscala/tree/master/exercises/binary-search-tree
- https://github.com/exercism/xswift/tree/master/exercises/binary-search-tree

**Comment by laurauzcategui** [comment]
Hi @behrtam @kytrinyx, 
 
Could you do quick review ? 

Cheers,

**Comment by laurauzcategui** [comment]
Also @behrtam , 

I think as I rebased the upstream, I brought your changes from there to my branch. I hope this is not a problem. Just wanted to update my branch with latest changes from master.

File: exercises/binary-search-tree/example.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+class Node(object):
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
+
+
+class BST(object):
+    def __init__(self, root):
+        self.root = Node(root)
```
Comment (patricksjackson) [inline exercises/binary-search-tree/example.py]: Requiring an initial value for the BST seems like a very odd design choice...

File: exercises/binary-search-tree/example.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if node.right is None and val > node.value:
+            node.right = Node(val)
+        elif node.left is None and val < node.value:
+            node.left = Node(val)
+        elif val > node.value:
+            self.pre_insert(node.right, val)
+        else:
+            self.pre_insert(node.left, val)
+
+    def search(self, find_val):
```
Comment (patricksjackson) [inline exercises/binary-search-tree/example.py]: This search function is probably a little easier to read if you do something like this:
```
cur = self.root
while cur:
    if find_val == cur.value:
        return True
    cur = cur.left if find_val < cur.value else cur.right
return False
```

**Comment by patricksjackson** [comment]
@laurauzcategui You can clean up the extra git commits by resetting instead of rebasing. It's explained here - https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#resetting-master

You also need to add a few more tests. Right now you only ever test inserting things into the left nodes. You should at least add some tests to try to insert a more complex tree and values that are already in the tree.

**Comment by laurauzcategui** [comment]
Thanks @patricksjackson  will check it over the weekend :+1:

**Comment by kytrinyx** [comment]
@laurauzcategui Would you have a chance to finish this one up? We'd love to include it on the site.

**Comment by laurauzcategui** [comment]
Hi @kytrinyx yes, sure thing. I'll be able to take a look probably next week and over the weekend. 

Cheers,

**Comment by kabiir** [comment]
@kytrinyx Looks like issue was automatically closed, but I have a BST implementation ready in python2 I would like to include this exercise. Can you assign this issue to me and add hacktoberfest label too?

**Comment by kytrinyx** [comment]
@kabiir that's great. Please submit your pull request. It doesn't need to have an open issue or a hacktoberfest label in order to be counted towards your Hacktoberfest metrics.

**Pull request closed without merging**